src/pipeline/pose_base.py

```diff
@@ -0,0 +1,122 @@
+
+ 
+    def tf_interpreter(self):
+        return self._tfengine._tf_interpreter
+
+
+    def thumbnail(self, image=None, desired_size=None):
+        """Resizes original image as close as possible to desired size.
+        Preserves aspect ratio of original image.
+        Does not modify the original image.
+        :Parameters:
+        ----------
+        image : PIL.Image
+            Input Image for AI model detection.
+        desired_size : (width, height)
+            Size expected by the AI model.
+        :Returns:
+        -------
+        PIL.Image
+            Resized image fitting for the AI model input tensor.
+        """
+        assert image
+        assert desired_size
+        log.debug('input image size = %r', image.size)
+        thumb = image.copy()
+        w, h = desired_size
+        try:
+            # convert from numpy to native Python int type
+            # that PIL expects
+            if isinstance(w, np.generic):
+                w = w.item()
+                w = int(w)
+                h = h.item()
+                h = int(h)
+            thumb.thumbnail((w, h))
+        except Exception as e:
+            msg = (f"Exception in "
+                   f"PIL.image.thumbnail(desired_size={desired_size}):"
+                   f"type(width)={type(w)}, type(height)={type(h)}"
+                   f"\n{e}"
+                   )
+            log.exception(msg)
+            raise RuntimeError(msg)
+        log.debug('thmubnail image size = %r', thumb.size)
+        return thumb
+
+
+    def resize(self, image=None, desired_size=None):
+        """Pad original image to exact size expected by input tensor.
+        Preserve aspect ratio to avoid confusing the AI model with
+        unnatural distortions. Pad the resulting image
+        with solid black color pixels to fill the desired size.
+        Do not modify the original image.
+        :Parameters:
+        ----------
+        image : PIL.Image
+            Input Image sized to fit an input tensor but without padding.
+            Its possible that one size fits one tensor dimension exactly
+            but the other size is smaller than
+            the input tensor other dimension.
+        desired_size : (width, height)
+            Exact size expected by the AI model.
+        :Returns:
+        -------
+        PIL.Image
+            Resized image fitting exactly the AI model input tensor.
+        """
+        assert image
+        assert desired_size
+        log.debug('input image size = %r', image.size)
+        thumb = image.copy()
+        delta_w = desired_size[0] - thumb.size[0]
+        delta_h = desired_size[1] - thumb.size[1]
+        padding = (0, 0, delta_w, delta_h)
+        new_im = ImageOps.expand(thumb, padding)
+        log.debug('new image size = %r', new_im.size)
+        assert new_im.size == desired_size
+        return new_im
+
+
+    @abstractmethod
+    def execute_model(self, img):
+        '''
```

Comment:
@bhavikapanara Can you please document more thoroughly what this abstract method expects as parameters and what results implementing methods should return. Also, let's start using [type annotations](https://auth0.com/blog/typing-in-python/) where possible to better self-document the code. Especially in classes and methods that are API contracts for future extensions and community contributions.

Reply:
Done